servicetalk-http-api/src/main/java/io/servicetalk/http/api/HeaderUtils.java

```diff
@@ -535,7 +535,7 @@ protected final void initNext(CharSequence cookieHeaderValue) {
          * @return the next {@link HttpCookiePair} value for {@link #next()}, or {@code null} if all have been parsed.
          */
         private HttpCookiePair findNext(CharSequence cookieHeaderValue) {
-            int semiIndex = indexOf(cookieHeaderValue, ';', nextNextStart);
+            int semiIndex = nextDelimiter(cookieHeaderValue);
```

Comment:
Should we use the same approach instead of `indexOf` in all other cases in this class (lines 396, 442, 645)?